src/lorawan_mac.erl

```diff
@@ -125,17 +125,19 @@ check_link(DevAddr, FCnt) ->
         [L] ->
             case fcnt_gap(L#link.fcntup, FCnt) of
                 N when N < ?MAX_FCNT_GAP ->
-                    mnesia:dirty_write(links, L#link{fcntup=FCnt}),
+                    NewFCnt = (L#link.fcntup + N) band 16#FFFFFFFF,
```

Comment:
That's very clever. It took me a while to understand it, but I finally got it. :-)